File: ebakery-backend/src/main/java/pl/damianmrowinski/ebakerybackend/domain/entity/Order.java
package pl.damianmrowinski.ebakerybackend.domain.entity;

import java.util.Date;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

import lombok.Getter;
import org.hibernate.annotations.CreationTimestamp;

@Entity
@Table(name = "order")
@Getter
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String firstName;

    private String lastName;

    private String mail;

    private String country;

    private String city;

    private String street;

    private String zipCode;

    @CreationTimestamp
    private Date createdDate;

}

File: ebakery-frontend/src/app/shared/model/order.ts
export class Order {

  firstName: string;
  lastName: string;
  mail: string;
  country: string;
  city: string;
  street: string;
  zipCode: string;

  constructor(firstName: string,
              lastName: string,
              mail: string,
              country: string,
              city: string,
              street: string,
              zipCode: string) {
    this.firstName = firstName;
    this.lastName = lastName;
    this.mail = mail;
    this.country = country;
    this.city = city;
    this.street = street;
    this.zipCode = zipCode;
  }
}
